sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -261,18 +264,190 @@ def _is_json_content_type(content_type: str) -> bool:
     return False
 
 
+def _build_sentinel(setting: Union[str, Tuple[str, str]]) -> Tuple[str, str]:
+    try:
+        key, label = setting
+    except IndexError:
+        key = setting
+        label = EMPTY_LABEL
+    if "*" in key or "*" in label:
+        raise ValueError("Wildcard key or label filters are not supported for refresh.")
+    return key, label
+
+
+def _is_retryable_error(error: HttpResponseError) -> bool:
+    """Determine whether the service error should be silently retried after a backoff period, or raised.
+    Don't know what errors this applies to yet, so just always raising for now.
+    """
+    # 408: Request Timeout
+    # 429: Too Many Requests
+    # 500: Internal Server Error
+    # 504: Gateway Timeout
+    return error.status_code in [408, 429, 500, 504]  # Back off on throttling error
+
+
+class _RefreshTimer:
+    """
+    A timer that tracks the next refresh time and the number of attempts.
+    """
+
+    def __init__(self, **kwargs):
+        self._interval: int = kwargs.pop("refresh_interval", 30)
+        self._next_refresh_time: float = time.time() + self._interval
+        self._attempts: int = 1
+        self._min_backoff: int = (
+            kwargs.pop("min_backoff", 30) if kwargs.get("min_backoff", 30) <= self._interval else self._interval
+        )
+        self._max_backoff: int = (
+            kwargs.pop("max_backoff", 600) if kwargs.get("max_backoff", 600) <= self._interval else self._interval
+        )
+
+    def reset(self) -> None:
+        self._next_refresh_time = time.time() + self._interval
+        self._attempts = 1
+
+    def retry(self) -> None:
+        self._next_refresh_time = time.time() + self._calculate_backoff() / 1000
+        self._attempts += 1
+
+    def needs_refresh(self) -> bool:
+        return time.time() >= self._next_refresh_time
+
+    def _calculate_backoff(self) -> float:
+        max_attempts = 63
+        millisecond = 1000  # 1 Second in milliseconds
+
+        min_backoff_milliseconds = self._min_backoff * millisecond
+        max_backoff_milliseconds = self._max_backoff * millisecond
+
+        if self._attempts <= 1 or self._max_backoff <= self._min_backoff:
```

Comment:
Did you mean to compare it with the refresh interval?
```suggestion
        if self._attempts <= 1 or self._interval <= self._min_backoff:
            return self._interval * millisecond
```

Reply:
max_backoff should be the smaller of the two values. i.e. max_backoff is never larger than interval. The constructor sets max_backoff to interval if interval is smaller.